cmd/influxd/generate/generator.go

```diff
@@ -0,0 +1,161 @@
+package generate
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sync"
+
+	"github.com/influxdata/influxdb/cmd/influxd/generate/internal/shard"
+	"github.com/influxdata/influxdb/kit/errors"
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/pkg/data/gen"
+	"github.com/influxdata/influxdb/pkg/limiter"
+	"github.com/influxdata/influxdb/storage"
+	"github.com/influxdata/influxdb/tsdb"
+	"github.com/influxdata/influxdb/tsdb/tsi1"
+	"github.com/influxdata/influxdb/tsdb/tsm1"
+)
+
+type Generator struct {
+	sfile *tsdb.SeriesFile
+
+	// Clean specifies whether to clean any of the data related files
+	Clean CleanLevel
+}
+
+func (g *Generator) Run(ctx context.Context, path string, gen gen.SeriesGenerator) ([]string, error) {
+	path = filepath.Join(path, "engine")
+	config := storage.NewConfig()
+
+	switch g.Clean {
+	case CleanLevelTSM:
+		if err := os.RemoveAll(path); err != nil {
+			return nil, err
+		}
+
+	case CleanLevelAll:
+		if err := os.RemoveAll(path); err != nil {
+			return nil, err
+		}
+	}
+
+	g.sfile = tsdb.NewSeriesFile(config.GetSeriesFilePath(path))
+	if err := g.sfile.Open(ctx); err != nil {
+		return nil, err
+	}
+	defer g.sfile.Close()
+	g.sfile.DisableCompactions()
+
+	ti := tsi1.NewIndex(g.sfile, config.Index, tsi1.WithPath(config.GetIndexPath(path)))
+	if err := ti.Open(ctx); err != nil {
+		return nil, fmt.Errorf("error opening TSI1 index: %s", err.Error())
+	}
+
+	files, err := g.writeShard(ti, gen, config.GetEnginePath(path))
+	if err != nil {
+		return nil, fmt.Errorf("error writing data: %s", err.Error())
+	}
+
+	ti.Compact()
+	ti.Wait()
+	if err := ti.Close(); err != nil {
+		return nil, fmt.Errorf("error compacting TSI1 index: %s", err.Error())
+	}
+
+	var (
+		wg   sync.WaitGroup
+		errs errors.List
+	)
+
+	parts := g.sfile.Partitions()
+	wg.Add(len(parts))
+	ch := make(chan error, len(parts))
+	limit := limiter.NewFixed(runtime.NumCPU())
+
+	for i := range parts {
+		go func(n int) {
+			limit.Take()
+			defer func() {
+				wg.Done()
+				limit.Release()
+			}()
+
+			p := parts[n]
+			c := tsdb.NewSeriesPartitionCompactor()
```

Comment:
I noticed that I don't think this https://github.com/influxdata/platform/pull/1599 made it over from Platform to InfluxDB. @benbjohnson can you check on that?

Reply:
@e-dard Yeah, I can take a look.

Reply:
Added PR: https://github.com/influxdata/influxdb/pull/12780